### DISCUSSION TITLE
CardsGrid and Grid/Col

### DISCUSSION THREAD

**Discussion by cstrat** [mantinedev/mantine#98] [Q&A] [answered]
I previously had used `<CardsGrid>` to build a UI with a number of settings panels.
I've since re-built it using `<Grid>` however I have had to put a `<Paper>` element with `style={{ height: '100%' }}` into each `<Col>`.

It would be nice if we could compose the `<Col>` as a `<Paper>` element.

I know the docs have an example of wrapping `<Col>` but wouldn't it be nice to be able to have `<Col component={Paper}>` and then pass in some props for the paper component. Maybe even have it automatically set height of whatever is being composed to 100%...

All of that could be achieved by wrapping the Col, but as it stands now the `<Grid>` isn't doing much beyond flex-box stuff.

**Comment by rtivital** [comment, marked as answer]
Hmm, I think that adding `component` prop to Col is not a good idea as this component will have to support style and className props and this will resolve in inconsistent behavior with custom components. I think `style={{ height: '100%' }}` is the best possible solution which is easy to understand and implement as in your case columns almost always will be used in a loop and you'll have to set it only once.
